contracts/l1/Bridge.sol

```diff
@@ -248,10 +268,11 @@ contract Bridge is IBridge, VersionedInitializable {
      * @param l1AToken token address
      * @param l2Token token address
      **/
-    function _approveToken(address l1AToken, uint256 l2Token)
-        internal
-        onlyValidL2Address(l2Token)
-    {
+    function _approveToken(
+        address l1AToken,
+        uint256 l2Token,
+        uint256 ceiling
+    ) internal onlyValidL2Address(l2Token) {
```

Comment:
Do you need this here? It is only called in 1 place, which in turn gets called on initialize() that already does that validation.
If you think it is really important to validate on this level, probably is not necessary to validate on initialize(), but I believe is better there, and fail fast

Reply:
on `initialize()`, we only validate the `l2Bridge` address but not the provided addresses for static_a_tokens on L2. That's why we verify the address of each at the moment of approval, which in return will make initialization fail if a wrong address was provided.

Reply:
True, I read it wrong